File: Day 1. Task 1.py
//Objective
//In this challenge, we practice calculating quartiles. Check out the Tutorial tab for learning materials and an instructional video!

//Task
//Given an array, X, of n integers, calculate the respective first quartile (Q1), second quartile (Q2), and third quartile (Q3). It is guaranteed that Q1, Q2, and Q3 are integers.

//Python3 solution:

import statistics as st

n = int(input())
data = list(map(int, input().split()))
freq = list(map(int, input().split()))

s = []
for i in range(n):
s += [data[i]] * freq[i]
N = sum(freq)
s.sort()

if n%2 != 0:
q1 = st.median(s[:N//2])
q3 = st.median(s[N//2+1:])
else:
q1 = st.median(s[:N//2])
q3 = st.median(s[N//2:])

ir = round(float(q3-q1), 1)
print(ir)

File: Day 1. Task 1.R
#Objective
#In this challenge, we practice calculating quartiles. Check out the Tutorial tab for learning materials and an instructional video!

#Task
#Given an array, X, of n integers, calculate the respective first quartile (Q1), second quartile (Q2), and third quartile (Q3). It is guaranteed that Q1, Q2, and Q3 are integers.

#R solution:

scanned <- scan(file("stdin"), skip = 1, quiet = TRUE)
aftersorted <- sort(scanned)
numered <- length(aftersorted)
lowerEndIndex <- floor(numered / 2)
upperStartIndex <- ceiling(numered / 2) + 1
lowerHalf <- aftersorted[1 : lowerEndIndex]
upperHalf <- aftersorted[upperStartIndex : numered]
q1 <- median(lowerHalf)
q2 <- median(scanned)
q3 <- median(upperHalf)
cat(q1, "\n")
cat(q2, "\n")
cat(q3)
